### PR TITLE
fix: added gitaly cgroups init image to package

### PR DESCRIPTION
**Pull request by JoeHCQ1**
## Description

Downstream we use this image to modify cgroups on gitaly nodes to improve resiliency. This isn't in the package right now, so we're having to add it to a local package so it is available. By putting it here in the original package we ensure the version always matches the rest of the install, and save others who do similar work time later.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-package-gitlab/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: values/upstream-values.yaml [hunk 1]
```diff
@@ -22,6 +22,11 @@ gitlab:
     image:
       repository: registry.gitlab.com/gitlab-org/build/cng/gitaly
       tag: v17.5.2
+    cgroups:
+      initContainer:
+        image:
+          repository: registry.com/gitlab-org/build/cng/gitaly-init-cgroups
```
Comment (zachariahmiller) [inline values/upstream-values.yaml:28]: these are all supposed to be registry.gitlab.com

**Review by zachariahmiller — CHANGES_REQUESTED** [review]

**Review by zachariahmiller — COMMENTED** [review]
Per the discussion on getting this working you had posted about in slack, shouldnt securityContext need to be overridden in the common values for this even if its not being enabled so it works with uds with minimum configuration?

**Comment by JoeHCQ1** [comment]
> Per the discussion on getting this working you had posted about in slack, shouldnt securityContext need to be overridden in the common values for this even if its not being enabled so it works with uds with minimum configuration?

Depends on the level of support we want to provide in this package for the custom init. Having the image available allows me to override things downstream at the bundle level until it works. We could backport the configuration settings too, but as it's broken again in my cluster for some reason, I'll have to wait on that changeset.

**Comment by JoeHCQ1** [comment]
From before Thanksgiving break, note from reviewer in slack:

> left a couple comments. Also based on what that container actually has on it it seems like we might not actually need this dedicated upstream only container, since the [source script](https://gitlab.com/gitlab-org/build/CNG/-/blob/master/gitaly-init-cgroups/Dockerfile?ref_type=heads) should be in the gitaly container already.

**Comment by JoeHCQ1** [comment]
> From before Thanksgiving break, note from reviewer in slack:
> 
> > left a couple comments. Also based on what that container actually has on it it seems like we might not actually need this dedicated upstream only container, since the [source script](https://gitlab.com/gitlab-org/build/CNG/-/blob/master/gitaly-init-cgroups/Dockerfile?ref_type=heads) should be in the gitaly container already.

The script is there, but if swapped in for the gitaly init container, it errors out. They're relying on the [docker-build-time `CMD`](https://gitlab.com/gitlab-org/build/CNG/-/blob/master/gitaly-init-cgroups/Dockerfile?ref_type=heads#L13) rather than setting a command themselves. Therefore, the normal gitaly container just starts up gitaly.

**Comment by JoeHCQ1** [comment]
> > Per the discussion on getting this working you had posted about in slack, shouldnt securityContext need to be overridden in the common values for this even if its not being enabled so it works with uds with minimum configuration?
> 
> Depends on the level of support we want to provide in this package for the custom init. Having the image available allows me to override things downstream at the bundle level until it works. We could backport the configuration settings too, but as it's broken again in my cluster for some reason, I'll have to wait on that changeset.

The securityContext is now overridden.

**Review by robmcelvenny — APPROVED** [review]
lgtm!

**Comment by zachariahmiller** [comment]
@JoeHCQ1 The required exception is missing from this. It needs to be in the config chart, templated based on a value, which defaults to `false`

**Comment by JoeHCQ1** [comment]
ah yes, that exception is probably in my nutanix bundle but it'd be needed here. Good catch.

**Review by robmcelvenny — APPROVED** [review]
LGTM

**Review by zachariahmiller — APPROVED** [review]

**Pull request merged**